src/modules/simulation/gz_bridge/GZMixingInterfaceMotor.cpp

```diff
@@ -0,0 +1,123 @@
+ *	distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *	used to endorse or promote products derived from this software
+ *	without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "GZMixingInterfaceMotor.hpp"
+
+bool GZMixingInterfaceMotor::init(const std::string &model_name)
+{
+
+	std::string motor_speed_topic = "/model/" + model_name + "/command/motor_speed";
+
+	if (!_node.Subscribe(motor_speed_topic, &GZMixingInterfaceMotor::motorSpeedCallback, this)) {
+		PX4_ERR("failed to subscribe to %s", motor_speed_topic.c_str());
+		return false;
+	}
+
+	std::string actuator_topic = "/model/" + model_name + "/command/motor_speed";
+	_actuators_pub = _node.Advertise<gz::msgs::Actuators>(actuator_topic);
+
+	if (!_actuators_pub.Valid()) {
+		PX4_ERR("failed to advertise %s", actuator_topic.c_str());
+		return false;
+	}
+
+	_wheel_encoders_pub.advertise();
+
+	ScheduleNow();
+
+	return true;
+}
+
+bool GZMixingInterfaceMotor::updateOutputs(bool stop_motors, uint16_t outputs[MAX_ACTUATORS], unsigned num_outputs,
+		unsigned num_control_groups_updated)
+{
+	unsigned active_output_count = 0;
+
+	for (unsigned i = 0; i < num_outputs; i++) {
+		if (_mixing_output.isFunctionSet(i)) {
+			active_output_count++;
+
+		} else {
+			break;
+		}
+	}
+
+	if (active_output_count > 0) {
+		gz::msgs::Actuators rotor_velocity_message;
+		rotor_velocity_message.mutable_velocity()->Resize(active_output_count, 0);
+
+		for (unsigned i = 0; i < active_output_count; i++) {
+			float output_scaler = 100.0f;
+			float scaled_output = (float)outputs[i] - output_scaler;
+			rotor_velocity_message.set_velocity(i, scaled_output);
+		}
+
+
+		if (_actuators_pub.Valid()) {
+			return _actuators_pub.Publish(rotor_velocity_message);
+		}
+	}
+
+	return false;
+}
+
+void GZMixingInterfaceMotor::Run()
+{
+	pthread_mutex_lock(&_node_mutex);
+	_mixing_output.update();
+	_mixing_output.updateSubscriptions(false);
+	pthread_mutex_unlock(&_node_mutex);
+}
+
+void GZMixingInterfaceMotor::motorSpeedCallback(const gz::msgs::Actuators &actuators)
```

Comment:
This should IMO be inside the GzBridge itself, since the mixing interface is specific to outputs

Reply:
I followed the architecture of GZMixingInterfaceESC when doing it, would you suggest I move both into GZBridge?

Reply:
What is the reason that you cannot use the GZMixingInterfaceESC?

Reply:
I really follows https://github.com/PX4/PX4-Autopilot/blob/d25938698715dbb6a10e0891471c33a08687bf85/src/modules/simulation/gz_bridge/GZMixingInterfaceESC.cpp#L109-L125 which outputs ESC commands and gets `esc_status` back. This one gives wheel commands and gets `wheel_encoders` feedback. I'm not a 100% sure why the separate wheel interface is necessary but I was assuming the ESC commands go to a propeller model in gazebo and are not suitable for a wheel. Correct me if I'm wrong, I don't understand this part.

Reply:
@MaEtUgR I think you are right. But then is the wheelmixer a complete copy of the esc? For me there is nothing different other than changing the scale